toy-project/apis/getFollowList.ts

```diff
@@ -0,0 +1,47 @@
+import { FollowUserType, followList } from '@/types/types';
+import axios from 'axios';
+import { SetterOrUpdater } from 'recoil';
+
+const PER_PAGE = 100;
+export const getFollowingList = async (setFollowings: SetterOrUpdater<followList[]>, tokenValue: string) => {
+  try {
+    const followingData = await axios.get(`${process.env.NEXT_PUBLIC_BASE_URL}/user/following?per_page=${PER_PAGE}`, {
+      headers: {
+        Accept: 'application/vnd.github+json',
+        Authorization: `Bearer ${tokenValue}`,
+      },
+    });
+
+    const data = followingData.data;
+
+    const followingList = data.map((user: FollowUserType) => {
+      const { id, login, avatar_url } = user;
+      return { id, login, avatar_url };
+    });
+
+    setFollowings(followingList);
+  } catch {
+    console.error('error');
+  }
+};
+
+export const getFollowerList = async (setFollowers: SetterOrUpdater<followList[]>, tokenValue: string) => {
+  try {
+    const followerData = await axios.get(`${process.env.NEXT_PUBLIC_BASE_URL}/user/followers?per_page=${PER_PAGE}`, {
+      headers: {
+        Accept: 'application/vnd.github+json',
+        Authorization: `Bearer ${tokenValue}`,
+      },
+    });
+    const data = followerData.data;
+
+    const followerList = data.map((user: FollowUserType) => {
+      const { id, login, avatar_url } = user;
+      return { id, login, avatar_url };
+    });
+
+    setFollowers(followerList);
+  } catch {
+    console.error('error');
+  }
+};
```

Comment:
현재 이렇게 recoil 의 setter 를 받아 사용하도록 되어있는 상황에서는 상위에서 항상 recoil과 함께 사용해야 하는 의존성이 생겨서 함수의 사용 방식이 특정 방법으로 강제돼요. 그런데 그럴 필요가 있을까요? 만약 recoil이 아닌 useState를 사용하도록 변경된다면 어떻게 될까요?

그리고 이 함수는 한번에 두 가지 일을 하고 있어요.
1. 서버에서 팔로워 목록을 불러온다.
2. Follower 상태에 값을 설정한다.

getFollowerList 함수를 서버에서 팔로워 목록을 불러오는 역할만 하게 하고, 값 설정은 이 함수를 사용하는 쪽에서 하면 이 함수를 더 유연하게 사용할 수 있어요.

```suggestion
export const getFollowerList = async (setFollowers: SetterOrUpdater<followList[]>, tokenValue: string) => {
    const followerData = await axios.get(`${process.env.NEXT_PUBLIC_BASE_URL}/user/followers?per_page=${PER_PAGE}`, {
      headers: {
        Accept: 'application/vnd.github+json',
        Authorization: `Bearer ${tokenValue}`,
      },
    });
    const data = followerData.data;

    const followerList = data.map((user: FollowUserType) => {
      const { id, login, avatar_url } = user;
      return { id, login, avatar_url };
    });

    return followerList;

};
```

Reply:
(건영이가 써준 to-be의 인자에 setFollowers는 빠지는걸 의도한걸거에요!)

```tsx
export const getFollowerList = async (tokenValue: string) => { ... }
```